src/store/modules/DownloadModule.ts

```diff
@@ -93,24 +93,19 @@ export const DownloadModule = {
     },
 
     mutations: {
-        updateDownload(state: State, update: UpdateObject) {
-            const newDownloads = [...state.allDownloads];
-            const index = newDownloads.findIndex((old: DownloadProgress) => {
-                return old.assignId === update.assignId;
-            });
-
-            if (index === -1) {
-                // The DownloadProgress by the ID from the update wasn't found at all.
-                console.warn(`Couldn\'t find DownloadProgress object with assignId ${update.assignId}.`);
-                return;
+        remove(state: State, assignId: number) {
+            const index: number = indexOfDownloadProgress(state.allDownloads, assignId);
```

Comment:
Nitpick: personally I would omit the `: number` here as the return type of `indexOfDownloadProgress` is strongly typed and the type of `index` in unlikely to cause confusion in the context of this mutator. Granted, it doesn't really hurt either but for me it seems like a distraction when reading this code.